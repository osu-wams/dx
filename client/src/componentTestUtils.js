import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
