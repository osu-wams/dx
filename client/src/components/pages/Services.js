import React, { Component } from 'react';
import PageTitle from '../layout/PageTitle';

export default class Services extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return (
      <div data-testid="services-page">
        <PageTitle title="Tools and Services" />
        <p>Services {api} </p>
      </div>
    );
  }
}
