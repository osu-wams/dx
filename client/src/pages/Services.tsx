import React, { FC, useState } from 'react';
import PageTitle from '../ui/PageTitle';
import ServiceCategories from '../features/ServiceCategories';

type api = {};

const Services: FC<{ initial?: {} }> = ({ initial = null }) => {
  const [api, setApi] = useState(initial);

  return (
    <div data-testid="services-page">
      <PageTitle title="Tools and Services" />
      <ServiceCategories />
      <p>Services {api} </p>
    </div>
  );
};

export default Services;
