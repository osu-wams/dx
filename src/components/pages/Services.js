import React, { Component } from 'react';

export default class Services extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return <div>Services {api} </div>;
  }
}
