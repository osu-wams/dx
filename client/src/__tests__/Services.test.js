import React from 'react';
import { render } from '../componentTestUtils';
import Services from '../components/pages/Services';

test('renders', () => {
  const { container } = render(<Services />);
  expect(container.firstChild).toMatchSnapshot();
});
