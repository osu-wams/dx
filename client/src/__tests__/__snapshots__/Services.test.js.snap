// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  data-testid="services-page"
>
  <div
    style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
  >
    <h2>
      Tools and Services
    </h2>
  </div>
  <p>
    Services 
    Api
     
  </p>
</div>
`;
